### DISCUSSION TITLE
Resizing transition using cache placholder

### DISCUSSION THREAD

**Discussion by leondeklerk** [coil-kt/coil#1490] [Q&A]
Hi,
For my application I currently have the following logic:

I have a list with images that, when an element is clicked, opens a dialog with a new ImageView. First, the dialog ImageView is resized to match the location and size of the list item, then `.load(..)` is called on that view with the cache key of the list item as a placeholder. Upon loading the image in the new view, a listener creates a new Android TransitionSet, adds it to the TransitionManager, resizes the imageView to match its parent, and changes `scaleType` from `CENTER_CROP` to `FIT_CENTER`.

See the following code snippet which achieves the transition:

```kotlin
val imageView = binding.imageView
val rect = startData.rect
imageView.scaleType = ImageView.ScaleType.CENTER_CROP
imageView.requestNewSize(rect.width(), rect.height())
imageView.applyMargin(rect) // Extension function to set 4 margins based on rect
imageView.post {
    imageView.load(uri) {
        placeholderMemoryCacheKey(cacheKey) // cacheKey comes from the list imageView
        size(binding.layout.width, binding.layout.height)
        listener(onSuccess = { _, _ ->
            val transition = TransitionSet()
            transition
                .addTransition(ChangeClipBounds())
                .addTransition(ChangeTransform())
                .addTransition(ChangeBounds())
                .addTransition(ChangeImageTransform())
                .addListener(/**listeners**/)

             TransitionManager.beginDelayedTransition(binding.imageContainer, transition)

             imageView.scaleType = ImageView.ScaleType.FIT_CENTER
             imageView.layoutParams = (imageView.layoutParams as ViewGroup.MarginLayoutParams).apply {
                 marginStart = 0
                 marginEnd = 0
                 topMargin = 0
                 bottomMargin = 0
                 width = MATCH_PARENT
                 height = MATCH_PARENT
             }
         })
    }
}
```

Now I really want to achieve the following:
- Once the placeholder is loaded it is used for the transition without waiting for the full image to load.
- Whenever the full image is loaded, it replaces the placeholder in the transition (or is just set if the transition already ended)
- Achieve this with logic in the Coil loading chain/custom transition.

So basically the question is if this is possible within Coil and if it is, what the general elements are that I would need.
(I don't expect a full solution, just happy with any help I can get)
